deepchem/models/molgan.py

```diff
@@ -0,0 +1,258 @@
+from typing import List, Tuple
+
+import tensorflow as tf
+from deepchem.feat.molecule_featurizers.molgan_featurizer import GraphMatrix
+from deepchem.models import WGAN
+from deepchem.models.layers import MolGANEncoderLayer
+from tensorflow import keras
+from tensorflow.keras import layers
+
+
+class BasicMolGANModel(WGAN):
+  """
+  Model for de-novo generation of small molecules based on work of Nicola De Cao et al. [1]_.
+  Utilizes WGAN infrastructure; uses adjacency matrix and node features as inputs.
+  Inputs need to be one-hot representation.
+
+  Examples
+  --------
+  import deepchem as dc
```

Comment:
Could you format this as follows:

```
Examples
--------
>>> import deepchem as dc
>>> from deepchem.imports import BasicMolGANModel as MolGAN
...
```

That will let the docs render the example code :)

Reply:
There is problem with this as doctest fails due to not expecting values when model is being trained.
I have tried to solve it but no matter what I tried it did not work as training is variable.

Reply:
If you want it to be formatted as code but not run as a doctest, you can do this:

```python
>>>
>> import deepchem as dc
>> from deepchem.imports import BasicMolGANModel as MolGAN
...
```